src/java/org/jivesoftware/openfire/net/DNSUtil.java

```diff
@@ -63,21 +63,29 @@
     }
 
     /**
-     * Returns a sorted list of host names and ports that the specified XMPP domain
-     * can be reached at for server-to-server communication. A DNS lookup for a SRV
-     * record in the form "_xmpp-server._tcp.example.com" is attempted, according
-     * to section 14.4 of RFC 3920. If that lookup fails, a lookup in the older form
-     * of "_jabber._tcp.example.com" is attempted since servers that implement an
-     * older version of the protocol may be listed using that notation. If that
-     * lookup fails as well, it's assumed that the XMPP server lives at the
-     * host resolved by a DNS lookup at the specified domain on the specified default port.<p>
+     * Returns a sorted list of host names and ports that the specified XMPP
+     * domain can be reached at for server-to-server communication.
+     *
+     * DNS lookups for a SRV records in the form "_xmpp-server._tcp.example.com"
+     * and "_xmpps-server._tcp.example.com" are attempted, in line with section
+     * 3.2 of XMPP Core and XEP-0368.
+     *
+     * If those lookup fail to provide any records, a lookup in the older form
+     * of "_jabber._tcp.example.com" is attempted since servers that implement
+     * an older version of the protocol may be listed using that notation.
+     *
+     * If that lookup fails as well, it's assumed that the XMPP server lives at
+     * the host resolved by a DNS A lookup at the specified domain on the
+     * specified default port.<p>
      *
      * As an example, a lookup for "example.com" may return "im.example.com:5269".
      *
      * @param domain the domain.
      * @param defaultPort default port to return if the DNS look up fails.
-     * @return a list of  HostAddresses, which encompasses the hostname and port that the XMPP
-     *      server can be reached at for the specified domain.
+     * @return a list of  HostAddresses, which encompasses the hostname and port
+     *         that the XMPP server can be reached at for the specified domain.
+     * @see <a href="https://tools.ietf.org/html/rfc6120#section-3.2">XMPP CORE</a>
+     * @see <a href="https://xmpp.org/extensions/xep-0368.html">XEP-0368</a>
      */
     public static List<HostAddress> resolveXMPPDomain(String domain, int defaultPort) {
```

Comment:
This looks generally good, but if the server has TLS disabled we should probably not perform the "_xmpps-server" lookup.